### PR TITLE
`azurerm_key_vault_certificate`: fix `createCertificate` lost `http.Response` which is used to check if should recover this certificate

### PR DESCRIPTION
**Pull request by wuxu92**
should fixes: #23177.

the failed `TestAccKeyVaultCertificate_basicGenerateUnknownIssuer` should not related to this PR. it has been failed for some time. I'll submit a separate PR(#23214)to fix it.

image

### CODE REVIEW

File: internal/services/keyvault/key_vault_certificate_resource.go [hunk 1]
```diff
@@ -512,11 +514,11 @@ func resourceKeyVaultCertificateCreate(d *pluginsdk.ResourceData, meta interface
 		if err != nil {
 			if meta.(*clients.Client).Features.KeyVault.RecoverSoftDeletedCerts && utils.ResponseWasConflict(newCert.Response) {
 				if err = recoverDeletedCertificate(ctx, d, meta, *keyVaultBaseUrl, name); err != nil {
-					return err
+					return fmt.Errorf("recover deleted certificqate: %+v", err)
```
Comment (stephybun) [inline internal/services/keyvault/key_vault_certificate_resource.go:517]:
```suggestion
					return fmt.Errorf("recover deleted certificate: %+v", err)
```

File: internal/services/keyvault/key_vault_certificate_resource.go [hunk 2]
```diff
@@ -528,12 +530,12 @@ func resourceKeyVaultCertificateCreate(d *pluginsdk.ResourceData, meta interface
...
```
Comment (stephybun) [inline internal/services/keyvault/key_vault_certificate_resource.go:533]:
```suggestion
					return fmt.Errorf("recover deleted certificate: %+v", err)
```

Comment (wuxu92) [reply internal/services/keyvault/key_vault_certificate_resource.go]: updated.

**Comment by itdevops-channelcapital** [comment]
Hi @wuxu92 and @stephybun,
How is this PR tracking? Seems to be just about done. Would be great to have this fix in the next release.

**Review by stephybun — CHANGES_REQUESTED** [review]
@wuxu92 could you please add an acceptance test that reproduces the failure described in the issue so that we can verify that this fixes it?

**Review by stephybun — COMMENTED** [review]
Thanks @wuxu92 LGTM 👍

**Review by stephybun — APPROVED** [review]

**Pull request merged**